ssz.go

```diff
@@ -128,6 +138,9 @@ func HashTreeRoot(val interface{}) ([32]byte, error) {
 	if val == nil {
 		return [32]byte{}, errors.New("untyped nil is not supported")
 	}
+	if b, ok := val.(bitfield.Bitvector4); ok {
```

Comment:
why is bitvector a special case?

Reply:
You're right - it doesn't make much sense. I instead modified `HashTreeRootBitfield` to take in a generic interface and do a type assertion.